src/SimpleThings/EntityAudit/AuditConfiguration.php

```diff
@@ -27,15 +27,49 @@
 
 class AuditConfiguration
 {
-    private $prefix = '';
-    private $suffix = '_audit';
-    private $revisionFieldName = 'rev';
-    private $revisionTypeFieldName = 'revtype';
-    private $revisionTableName = 'revisions';
-    private $auditedEntityClasses = array();
-    private $globalIgnoreColumns = array();
-    private $currentUsername = '';
-    private $revisionIdFieldType = 'integer';
+    private $auditedEntityClasses;
+    private $globalIgnoreColumns;
+    private $tablePrefix;
+    private $tableSuffix;
+    private $revisionTableName;
+    private $revisionFieldName;
+    private $revisionTypeFieldName;
+    private $revisionIdFieldType;
+    private $usernameCallable;
+
+    /**
+     * @param array $auditedEntityClasses
+     * @param array $globalIgnoreColumns
+     * @param $tablePrefix
+     * @param $tableSuffix
+     * @param $revisionTableName
+     * @param $revisionFieldName
+     * @param $revisionTypeFieldName
+     * @param $revisionIdFieldType
+     * @param callable $usernameCallable
+     */
+    public function __construct(
+        array $auditedEntityClasses = array(),
+        array $globalIgnoreColumns = array(),
+        $tablePrefix = '',
+        $tableSuffix = '_audit',
+        $revisionTableName = 'revisions',
+        $revisionIdFieldType = 'integer',
+        $revisionFieldName = 'rev',
+        $revisionTypeFieldName = 'revtype',
+        callable $usernameCallable = null
```

Comment:
@bendavies "Callbacks can be denoted by callable type hint as of PHP 5.4."  -> http://php.net/manual/en/language.types.callable.php